[need help] updating changelog for v0.8

choldgraf opened the pull request:
Since we are ready to release v0.8 (#1054) this PR updates the changelog for new features.

**It needs help from the jupyterhub team to fill in missing pieces!**. Please do this either via "suggested edits" in the GitHub UI, or by pushing directly to my branch.

I've marked areas where we need more information with "TODO".

Once we get it filled out, we can merge and get a move-on with the new version!

cc @jupyterhub/jupyterhubteam @jupyterhub/binder-team 

# Things I could use help with

- Documenting any upgrade tips / warnings for people coming from v0.7
- Running "upgrade tests" from v0.7 (not sure what this means but @minrk has it in #1054)
- Documenting any breaking changes
- Pushing a v0.8 image to Dockerhub (https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/master/CONTRIBUTING.md#push-built-images-to-dockerhub--bump-version)

manics commented on CHANGELOG.md:
```diff
@@ -5,13 +5,84 @@ Releases are named after famous [Cricket](https://en.wikipedia.org/wiki/Cricket)
+## [Unreleased](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/compare/0.8.0..master)
 
 ### Breaking changes
 
```
```suggestion
- Github organisation OAuth: `auth.github.org_whitelist` has been renamed to `auth.github.orgWhitelist`

```
See https://github.com/jupyterhub/zero-to-jupyterhub-k8s/issues/1118. This is a potential security hole for anyone who doesn't realise this since if no org is whitelisted any GitHub user can login

choldgraf replied:
woot! thanks for the suggestion!

manics commented:
`hub.extraConfig`: not clear to me whether the old format is deprecated:
- https://zero-to-jupyterhub.readthedocs.io/en/latest/advanced.html?highlight=extraconfig#hub-extraconfig says the old and new formats are supported
- https://zero-to-jupyterhub.readthedocs.io/en/latest/reference.html?highlight=extraconfig#custom only has the new format
- If you use the old format the hub logs say
```
hub.extraConfig should be a dict of strings,
but found a single string instead.

The keys can be anything identifying the
block of extra configuration.

Try this instead:

    hub:
      extraConfig:
        myConfig: |
```

consideRatio commented:
The new format allows for multiple extraConfig entries as a dict can be merged by Helm, while a string cant.

So, let's add a deprecation note in the changelog?

consideRatio commented on CHANGELOG.md:
```diff
@@ -5,13 +5,85 @@ Releases are named after famous [Cricket](https://en.wikipedia.org/wiki/Cricket)
+
+### New Features
+
+#### TODO: ITEM 1
```
- Better autoscaling support, as can be read about in http://z2jh.jupyter.org/en/latest/optimization.html

consideRatio commented on CHANGELOG.md:
```diff
@@ -5,13 +5,85 @@ Releases are named after famous [Cricket](https://en.wikipedia.org/wiki/Cricket)
+
+### Upgrading from 0.7
+
+TODO: Instructions
```
I think using `--force` should be recommended at all times, I don't remember if it is required but I think it may be.

choldgraf commented:
I added a few more features using @consideRatio's excellent summary from #841. I also added the contributor list. Edits / additions / subtractions / reorgs are welcome!

minrk commented:
@consideRatio 

> So, let's add a deprecation note in the changelog?

There is a deprecation note in the changelog when this was deprecated (0.6). What's changed in 0.8 is that users are now notified when they use the deprecated configuration. I updated the deprecation note in the image to clarify that the config is loaded and a note in the changelog that the deprecation warnings and validation errors are a new feature.

minrk commented:
I added a few more notes for things I found, and some upgrade and troubleshooting steps. I think we're in pretty good shape for making a beta release tomorrow.

consideRatio commented on CHANGELOG.md:
```diff
@@ -20,14 +20,37 @@ It contains new features, additional configuration options, and bug fixes.
 
+To upgrade your cluster:
+
+1. backup any persistent volumes and previous configuration, to be safe
```
I find it quite extreme to recommend backing up of PDs, do we really want to suggest this? It is a tricky thing to do.

choldgraf replied:
I feel like it's always safe to say "we recommend backing up XXX" as long as you don't imply that you **must** back up PDs. We don't want to create unnecessary work for people, but we also don't wanna be on the hook if something goes wrong and a person gets mad at us because we never mentioned it :-)

minrk replied:
We should clarify that this is only helm-managed PVs (just hub state), not user PVs which are not managed by the chart, and thus won't be deleted by an upgrade.

consideRatio commented on doc/source/google/step-zero-gcp.rst:
```diff
@@ -102,14 +102,54 @@ your google cloud account.
+    <https://cloud.google.com/compute/docs/machine-types>`_. See
+    the `preemptible node documentation
+    <https://cloud.google.com/compute/docs/instances/preemptible>`_ for more
+    information.
```
Related to this, is an issue that is not yet resolved I think: https://github.com/jupyterhub/kubespawner/issues/223

consideRatio replied:
In summary, I recall that whenever a user is evicted, as they are from preemptible node pools that run out of their usage time for some reason, they will not be able to login again without a hub-pod restart.

minrk replied:
I don't think that's something we should look at for this release. Do you think this means we should not recommend looking at preemptible nodes for now?

minrk replied:
commented out the recommendation for now

consideRatio replied:
Yeah, I would not recommend them until one of us have successfully utilized them with users that authenticate regularly etc. I ended up with trouble doing it.

consideRatio commented:
@minrk this is excellent!!

This LGTM!

minrk commented:
OK! Let's hit the button and make the beta. This PR updates the version in the docs to install the beta, so just building and publishing the tag should do it.

minrk commented:
ok, published 0.8.0-beta.1 (I forgot that helm doesn't accept Python semver-style 0.8.0b1)
